csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -56,24 +45,80 @@ class AllocationOrderInferencer : public IterVisitor {
   void handle(BinaryOp*) override;
   void handle(TernaryOp*) override;
   void handle(PadOp*) override;
+  void handle(ReductionOp*) override;
   // TODO: Add more propagation rules
-  // void handle(Reduction*) override;
   // void handle(LoadStoreOp*) override;
   // void handle(SqueezeOp*) override;
   // void handle(ExpandOp*) override;
 
  private:
-  // propagate allocation order from src to dst. Returns true when src has a
-  // recorded allocation order, false otherwise.
-  bool propagateAllocationOrder(TensorView* src, TensorView* dst) {
-    if (auto iter = alloc_order_map_.find(src);
+  // mapping allocation domain from producer to consumer without reduction
+  //
+  // e.g.
+  //   producer rfactor dom [r0', i0', i1', i2'] @ allocation order {0, 1, 3, 2}
+  //    |       alloc dom [r0', i0', i2', i1']
+  //    |
+  //    Operation
+  //    |
+  //    v
+  //   consumer rfactor dom [..., i0, ..., i1, ..., i2, ...]
+  //
+  // we construct allocation domain on producer, filtering out reduction, apply
+  // root domain map from producer to consumer.
+  //   [r0', i0', i2', i1'] -> [i0', i2', i1'] -> [i0, i2, i1]
+  // so the function would return [i0, i2, i1]
+  std::vector<IterDomain*> propagateAllocationDomain(
```

Comment:
```
propagateAllocationDomainFromProducerToConsumer
```

Reply:
Actually, it's a little confusing to have this and the below function both named as "propagate". This one doesn't update any state, but the other does update. Maybe like `getConsumerAllocationDomainFromProducer`.